src/internal/confluent/client.go

```diff
@@ -282,41 +282,35 @@ func (c *Client) createApiKey(ctx context.Context, resourceId string, serviceAcc
 
 	if err != nil {
 		//log -> response.Status
-		return nil, err
+		return models.ApiKey{}, err
```

Comment:
Any reason why we do away with pointers for the returned value? Just curious.

Reply:
I think pointer returns are a bit improper outside of constructors. You are extending the lifetime of the object beyond its own scope, which is a bit strange. Since apikey is a reasonably simple type (two strings), i think it makes the object lifecycle easier to understand.